core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -216,9 +216,15 @@ public static void toJson(TableMetadata metadata, JsonGenerator generator) throw
     // write properties map
     JsonUtil.writeStringMap(PROPERTIES, metadata.properties(), generator);
 
-    generator.writeNumberField(
-        CURRENT_SNAPSHOT_ID,
-        metadata.currentSnapshot() != null ? metadata.currentSnapshot().snapshotId() : -1);
+    if (metadata.currentSnapshot() != null) {
+      generator.writeNumberField(CURRENT_SNAPSHOT_ID, metadata.currentSnapshot().snapshotId());
+    } else {
+      if (metadata.formatVersion() >= 3) {
```

Comment:
I'd prefer we tie this to a constant,

MIN_NULL_CURRENT_SNAPSHOT_VERSION or something like that. I just want to have all of these tweaks in one place so we can see all the behavior changes in a constant list.

Reply:
I missed this comment, but I've added it 👍